docs/pages/base-ui/api/tab.json

```diff
@@ -1,11 +1,6 @@
 {
   "props": {
-    "action": {
-      "type": {
-        "name": "union",
-        "description": "func<br>&#124;&nbsp;{ current?: { focusVisible: func } }"
-      }
-    },
+    "action": { "type": { "name": "custom", "description": "ref" } },
```

Comment:
@oliviertassinari for these custom refs, it might be debatable if the new generation is actually better.

I'd say that `func | { current?: { focusVisible: func } }` was not very readable anyway, but still we are loosing information.

I can avoid using `refType` when the ref value is not an HTML element.

Reply:
Of change `refType` to have `refDOMNodeType` for the HTML elements and `refType(...)` that takes a param for the custom refs

```tsx
actions: refType({
  focusVisible: PropTypes.func,
})
```

Reply:
Agree, I think a ref implies access to a DOM node, which isn't the case here. It feels better as before.

https://mui.com/base-ui/react-tabs/components-api/#Tab-prop-action better than: https://deploy-preview-38917--material-ui.netlify.app/base-ui/react-tabs/components-api/#Tab-prop-action.

Reply:
> It feels like we are hitting a limit of TypeScript → prop-types → docs. It should really be:

That's the main goal of my architectural changes.
I created a layer that only analyses the AST trees and that should be usable by both and let both add the specificity they want.